### PR TITLE
Use correct type names in banning string methods.

### LINKED ISSUE #10032: "String" related rules in conformance-config.textproto are not being enforced
Somehow none of the rules for `string` in https://github.com/ampproject/amphtml/blob/master/build-system/conformance-config.textproto#L38 are being applied. `Array`, `Number` rules work just fine and get reported properly.

Steps to repro:
1- Find an string anywhere in AMP code.
2- Add `thatString.normalize()` or `thatString.includes('foo')` somewhere there.
3- Run `gulp check-types`
4- Note that it passes, but according to the rules, `normalize` or `includes` are disallowed.

This seems to happen regardless of whether the string in question is a constant, variable or even a fully typed param.

/to @erwinmombay 
/cc @cramforce @jridgewell 
Related to: https://github.com/ampproject/amphtml/pull/10016

### PR DESCRIPTION
**Pull request by cramforce**
Fixes #10032

### CODE REVIEW

File: build-system/conformance-config.textproto [hunk 1]
```diff
@@ -11,56 +11,58 @@ requirement: {
 
 requirement: {
   type: BANNED_PROPERTY_CALL
-  error_message: 'String.prototype.padStart is not allowed'
-  value: 'String.prototype.padStart'
+  error_message: 'string.prototype.padStart is not allowed'
```
Comment (jridgewell) [inline build-system/conformance-config.textproto]: Damn it Java.

Comment (cramforce) [inline build-system/conformance-config.textproto]: This is a bit of an ambiguous case. The way to get to the actual function in JS is to use `String.prototype.padStart`, but the name of the unboxed type is `string`

Comment (jridgewell) [inline build-system/conformance-config.textproto]: Nope, I'm still blaming it on Java. 😜

Comment (cramforce) [inline build-system/conformance-config.textproto]: OK. yes. This is Java's fault.

**Pull request merged**